src/sentry/rules/history/preview_strategy.py

```diff
@@ -72,7 +72,7 @@ def _transactions_from_groups_kwargs(
 
 
 def get_update_kwargs_for_groups(
-    dataset: Dataset,
+    dataset: Dataset | None,
```

Comment:
feels like the caller shouldn't pass `None` into here?  where is the `None` coming from?

yeah looks like the correct fix should be on preview.py:256 -- that shouldn't be using `.get(...)` which is what introduces the nasty nullability issue here